test/e2e/e2e_test.go

```diff
@@ -1252,6 +1253,185 @@ var _ = Describe("e2e non-vxlan external gateway and update validation", func()
 	})
 })
 
+// Validate pods can reach a network running in a container's looback address via
+// an external gateway running on eth0 of the container without any tunnel encap.
+// The traffic will get proxied through an annotated pod in the default namespace.
+// This test requires the infa to have shared gateway mode to be enabled.
```

Comment:
now since local gateway mode uses same topology as shared gateway mode, this should work in either mode